### ISSUE TITLE
torrent - assertion failed

### ISSUE THREAD

**Issue by centaur1** [miguelfreitas/twister-core#384]
version: 1.0.0.0
$Rev: 7680 $
file: 'libtorrent/src/torrent.cpp'
line: 3066
function: void libtorrent::torrent::piece_finished(int, int, uint32_t, int)
expression: !m_picker->have_piece(index)

stack:
1: ) [0x441d10]
2: ) [0x5ccf0a]
3: ) [0x5cee0b]
4: ) [0x5dbc20]
5: ) [0x46549e]
6: ) [0x474694]
7: ) [0x4383cc]
8: ) [0x43c1d1]
9: ) [0x581acf]
10: ) [0x430d8f]
11: 
12: clone

**Comment by centaur1** [comment]
Got it again.
Please include the following information:

version: 1.0.0.0
$Rev: 7680 $
file: 'libtorrent/src/torrent.cpp'
line: 3066
function: void libtorrent::torrent::piece_finished(int, int, boost::uint32_t, int)
expression: !m_picker->have_piece(index)

stack:
1: assert_fail(char const*, int, char const*, char const*, char const*)
2: libtorrent::torrent::piece_finished(int, int, unsigned int, int)
3: boost::detail::function::void_function_obj_invoker2<boost::_bi::bind_t<void, boost::_mfi::mf4<void, libtorrent::torrent, int, int, unsigned int, int>, boost::_bi::list5<boost::_bi::value<boost::shared_ptr<libtorrent::torrent> >, boost::_bi::value<int>, boost::arg<1>, boost::arg<2>, boost::_bi::value<int> > >, void, int, unsigned int>::invoke(boost::detail::function::function_buffer&, int, unsigned int)
4: libtorrent::torrent::on_piece_verified(int, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)>)
5: void boost::_mfi::mf3<void, libtorrent::torrent, int, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)> >::call<boost::shared_ptr<libtorrent::torrent>, int, libtorrent::disk_io_job const, boost::function<void (int, unsigned int)> >(boost::shared_ptr<libtorrent::torrent>&, void const*, int&, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)>&) const
6: void boost::_bi::list4<boost::_bi::value<boost::shared_ptr<libtorrent::torrent> >, boost::arg<1>, boost::arg<2>, boost::_bi::value<boost::function<void (int, unsigned int)> > >::operator()<boost::_mfi::mf3<void, libtorrent::torrent, int, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)> >, boost::_bi::list2<int&, libtorrent::disk_io_job const&> >(boost::_bi::type<void>, boost::_mfi::mf3<void, libtorrent::torrent, int, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)> >&, boost::_bi::list2<int&, libtorrent::disk_io_job const&>&, int)
7: boost::detail::function::void_function_obj_invoker2<boost::_bi::bind_t<void, boost::_mfi::mf3<void, libtorrent::torrent, int, libtorrent::disk_io_job const&, boost::function<void (int, unsigned int)> >, boost::_bi::list4<boost::_bi::value<boost::shared_ptr<libtorrent::torrent> >, boost::arg<1>, boost::arg<2>, boost::_bi::value<boost::function<void (int, unsigned int)> > > >, void, int, libtorrent::disk_io_job const&>::invoke(boost::detail::function::function_buffer&, int, libtorrent::disk_io_job const&)
8: libtorrent::completion_queue_handler(std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*)
9: void boost::asio::asio_handler_invoke<boost::_bi::bind_t<void, void (*)(std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*), boost::_bi::list1<boost::_bi::value<std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*> > > >(boost::_bi::bind_t<void, void (*)(std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*), boost::_bi::list1<boost::_bi::value<std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*> > >&, ...)
10: boost::asio::detail::completion_handler<boost::_bi::bind_t<void, void (*)(std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*), boost::_bi::list1<boost::_bi::value<std::__1::list<std::__1::pair<libtorrent::disk_io_job, int>, std::__1::allocator<std::__1::pair<libtorrent::disk_io_job, int> > >*> > > >::do_complete(boost::asio::detail::task_io_service*, boost::asio::detail::task_io_service_operation*, boost::system::error_code const&, unsigned long)
11: boost::asio::detail::task_io_service::do_run_one(boost::asio::detail::scoped_lock<boost::asio::detail::posix_mutex>&, boost::asio::detail::task_io_service_thread_info&, boost::system::error_code const&)
12: boost::asio::detail::task_io_service::run(boost::system::error_code&)
13: boost::asio::io_service::run(boost::system::error_code&)
14: libtorrent::aux::session_impl::main_thread()
15: boost::asio::detail::posix_thread::func<boost::_bi::bind_t<void, boost::_mfi::mf0<void, libtorrent::aux::session_impl>, boost::_bi::list1<boost::_bi::value<libtorrent::aux::session_impl*> > > >::run()
16: boost_asio_detail_posix_thread_function
17: _pthread_body
18: _pthread_body
19: thread_start

Abort trap: 6

**Comment by centaur1** [comment]
I usually see this when twisterd has recently started - e.g. within the first few minutes or hours.  It seems like when it is starting up the torrents/catching up with the torrents that it has the most issues. Particularly when you try to use it while it is coming up.  Once it has gone 6-8 hours without crashing, it seems to be much more stable.  For example I restarted it yesterday it around 9am, and it crashed three times before 10am (after which I restarted it each time0.  But then once it restarted and no one used it for a while once it started, it has been stable now for 36+ hours no matter what is done with it (within reason).
